src/Language/PureScript/Sugar/Names.hs

```diff
@@ -99,15 +99,12 @@ desugarImportsWithEnv externs modules = do
     exportedRefs f = M.fromList $ (, efModuleName) <$> mapMaybe f efExports
 
   updateEnv :: ([Module], Env) -> Module -> m ([Module], Env)
-  updateEnv (ms, env) m@(Module ss _ mn _ refs) =
-    case mn `M.lookup` env of
-      Just m' -> throwError . errorMessage $ RedefinedModule mn [envModuleSourceSpan m', ss]
```

Comment:
No need to check if a module already exists in the name environment anymore since it will be caught during `Make` and the error raised there.